docs/performance/optimization.rst

```diff
@@ -0,0 +1,595 @@
+   INNER JOIN factory_metadata ON filtered_device_data.factory_id=factory_metadata.factory_id;
+
+.. _filter-with-array-expressions:
+
+Use filters with array expressions when filtering on the output of UNNEST
+=========================================================================
+
+On denormalized data sets you may have records with an array of objects.
+
+You may want to unnest the array in a subquery or CTE and later filter on a
+property of the OBJECTs.
+
+The next statement will result in every row in the table (not filtered
+with other conditions) being read and unnested, to check if it meets
+the criteria on ``field1``.
+
+.. code:: sql
+
+   SELECT *
+   FROM (
+      SELECT UNNEST(my_array_of_objects) obj
+      FROM my_table
+   )
+   WHERE obj['field1'] = 1;
+
+However, CrateDB can do a lot better than this if we add an additional condition
+like this:
+
+.. code:: sql
+
+   SELECT *
+   FROM (
+      SELECT unnest(my_array_of_objects) obj
+      FROM my_table
+      WHERE 1 = ANY (my_array_of_objects['field1'])
+   ) AS subquery
+   WHERE obj['field1'] = 1;
+
+CrateDB leverages indexes to only unnest the relevant records from ``my_table``
+which can make a huge difference.
+
+.. _group-efficient-query-structure:
+
+******************************************
+ Efficient Query Structure and Constructs
+******************************************
+
+This section focuses on optimizing SQL logic by prioritizing efficient syntax
+and avoiding redundant operations.
+
+.. _only-sort-when-needed:
+
+Only sort data when needed
+==========================
+
+Indexing in CrateDB is optimized to support filtering and aggregations without
+requiring expensive defragmentation operations, but it is not optimized for
+sorting​.
+
+Maintaining a sorted index would slow down ingestion, that is why​ other
+analytical database systems like Cassandra and Redshift make similar
+trade-offs​.
+
+This means that when an ``ORDER BY`` operation is requested, the whole dataset
+needs to be loaded into the main memory on the relevant cluster node to be sorted.
+That is why it is important to not request ``ORDER BY`` operations when not actually
+needed, and most importantly, not on tables of large cardinalities without aggregating
+records beforehand. On the other hand, of course it is no problem to sort a few
+thousand rows in the final stage of a ``SELECT`` operation, but we need to avoid
+requesting sort operations over millions of rows.
+
+Consider leveraging filters and aggregations like ``max_by`` and ``min_by`` to
+limit the scope of ``ORDER BY`` operations, or
+avoid them altogether when possible.
+
+So for instance instead of:
+
+.. code:: sql
+
+   SELECT reading_time,reading_value
+   FROM device_data
+   WHERE reading_time BETWEEN '2024-01-01' AND '2025-01-01'
+   ORDER BY reading_time DESC
+   LIMIT 10;
+
+use:
+
+.. code:: sql
+
+   SELECT reading_time,reading_value
+   FROM device_data
+   WHERE reading_time BETWEEN '2024-12-20' AND '2025-01-01'
```

Comment:
From what I see, the time range is the only difference. The section's title is "Only sort data when needed", it reads more like a "don't select more data than you need" example. The example could just omit `ORDER BY` entirely, as basically the last sentence before the example suggests?